src/resolvers/git.cr

```diff
@@ -288,7 +361,19 @@ module Shards
         raise Error.new("Failed #{command} (#{message}). Maybe a commit, branch or file doesn't exist?")
       end
     end
-  end
 
-  register_resolver GitResolver
+    def report_version(version : Version) : String
+      git_version = parse_git_version(version)
+      if commit = git_version.commit
+        "#{git_version.value} at #{commit[0...7]}"
+      else
+        version.value
+      end
+    end
+
+    register_resolver "git", GitResolver
```

Comment:
Just checking, we are not losing the work done in #315 by registering the same resolver class for different keys, right?

Reply:
That algorithm works on the full URL, so no.